### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Ibrahim-Khan88**

### CODE REVIEW

File: src/main/java/com/loofi/auth/jwt/TokenFilter.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+public class TokenFilter extends OncePerRequestFilter {
+	@Autowired
+	private JwtUtils jwtutils;
+	private static final Logger logger = LoggerFactory.getLogger(TokenFilter.class);
+
+	@Override
+	protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+			throws ServletException, IOException {
+		try {
+			String jwtToken = getToken(request);
```
Comment (azomshahriar) [inline src/main/java/com/loofi/auth/jwt/TokenFilter.java:35]: Can't we make function more small?

File: src/main/java/com/loofi/auth/service/UserCredentialService.java [hunk 2]
```diff
@@ -1,35 +1,137 @@
...
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 @Service
-@RequiredArgsConstructor
 public class UserCredentialService {
   public static final String SHA_256 = "SHA-256";
-  private final UserCredentialRepository userCredentialRepository;
+
+  @Autowired
```
Comment (azomshahriar) [inline src/main/java/com/loofi/auth/service/UserCredentialService.java:40]: Remove autowiare and add require argument constructor

File: src/main/java/com/loofi/auth/service/UserCredentialService.java [hunk 3]
```diff
@@ -1,35 +1,137 @@
...
+    userCredential.setWrongCredCount(userCredential.getWrongCredCount() + 1);
+          if (userCredential.getWrongCredCount() == wrongCredentialCount) {
+        userCredential.setState(UserCredentialState.BLOCK);
+  }
+    userCredentialRepository.save(userCredential);
+  }
+
+  public String validateCredential(LoginRequest loginRequest) throws BlockCredException, InvalidCredException {
+    UserCredential userCredential = userCredentialRepository.findByUserIdentifier(loginRequest.getIdentifier()).orElseThrow(() -> new RuntimeException("Account Not Found"));
+    if(userCredential != null) {
```
Comment (azomshahriar) [inline src/main/java/com/loofi/auth/service/UserCredentialService.java:102]: can't we make method small and format the code

File: src/main/java/com/loofi/auth/webconfing/Webconfig.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+		http.cors().configurationSource(request -> new CorsConfiguration().applyPermitDefaultValues());
+		http.csrf().disable().exceptionHandling().authenticationEntryPoint(unauthorizedHandler).and()
+				.sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS).and().authorizeRequests()
+				.antMatchers("/api/check/auth", "/api/signup")
+				.permitAll()
+				.antMatchers(HttpMethod.POST,"/api/change/auth/password").hasAnyRole("MAKER", "CHECKER")
+				.antMatchers(HttpMethod.POST,"/api/user-info").hasAnyRole("MAKER", "CHECKER")
+				.antMatchers(HttpMethod.POST,"/api/change/auth/state").hasAnyRole("MAKER")
+				// .and().httpBasic();
```
Comment (azomshahriar) [inline src/main/java/com/loofi/auth/webconfing/Webconfig.java:45]: remove

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -1 +1,13 @@
 
+spring.datasource.username=postgres
+spring.datasource.password=ibrahim1
```
Comment (azomshahriar) [inline src/main/resources/application.properties:3]: remove name add general username

**Review by azomshahriar — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties (hunk 5 above)
Comment (Ibrahim-Khan88) [reply src/main/resources/application.properties:3]: What do you mean by general username?

File: src/main/java/com/loofi/auth/service/UserCredentialService.java (hunk 3 above)
Comment (Ibrahim-Khan88) [reply src/main/java/com/loofi/auth/service/UserCredentialService.java:102]: I think the function not looking so big. Maximum function wrap between 15-20 lines. But you can suggest me how can i make this small.

**Review by azomshahriar — APPROVED** [review]

**Pull request merged**